dashboard/lib/middlewares/redis_session_store.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Middlewares
+  class RedisSessionStore < ActionDispatch::Session::RedisStore
+    # Extends +ActionDispatch::Request::Session+ to allow identifying whether a session has been changed.
+    module SessionExtension
+      def changed?
+        @changed.present?
+      end
+
+      private def load_for_write!
```

Comment:
nit; please add a comment linking to https://github.com/rails/rails/blob/v6.1.7.7/actionpack/lib/action_dispatch/request/session.rb#L229